internal/venice-common/src/main/java/com/linkedin/venice/response/VeniceReadResponseStatus.java

```diff
@@ -1,19 +1,40 @@
 package com.linkedin.venice.response;
 
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+
 /**
- * Enumeration of response status codes for Venice read requests.
- * <p>
- * **Positive values** correspond to standard HTTP status codes and can be used directly in HTTP responses.
- * **Negative values** represent custom Venice-specific error codes.
- * <p>
- * For example, a status code of `200` indicates a successful read, while a status code of `-100` might indicate a specific Venice-related error.
+ * Defines response status codes for Venice read requests. This wrapper around {@link HttpResponseStatus} allows
+ * for the inclusion of custom status codes that extend beyond the standard HTTP status codes.
  */
-public class VeniceReadResponseStatus {
-  public static final int KEY_NOT_FOUND = -420;
-
-  public static final int OK = 200;
-  public static final int BAD_REQUEST = 400;
-  public static final int INTERNAL_ERROR = 500;
-  public static final int TOO_MANY_REQUESTS = 429;
-  public static final int SERVICE_UNAVAILABLE = 503;
+public enum VeniceReadResponseStatus {
+  KEY_NOT_FOUND(HttpResponseStatus.NOT_FOUND), OK(HttpResponseStatus.OK), BAD_REQUEST(HttpResponseStatus.BAD_REQUEST),
+  FORBIDDEN(HttpResponseStatus.FORBIDDEN), METHOD_NOT_ALLOWED(HttpResponseStatus.METHOD_NOT_ALLOWED),
+  REQUEST_TIMEOUT(HttpResponseStatus.REQUEST_TIMEOUT), TOO_MANY_REQUESTS(HttpResponseStatus.TOO_MANY_REQUESTS),
+  INTERNAL_SERVER_ERROR(HttpResponseStatus.INTERNAL_SERVER_ERROR),
+  SERVICE_UNAVAILABLE(HttpResponseStatus.SERVICE_UNAVAILABLE),
+  MISROUTED_STORE_VERSION(new HttpResponseStatus(570, "Misrouted request"));
+
+  private final HttpResponseStatus httpResponseStatus;
+
+  VeniceReadResponseStatus(HttpResponseStatus httpResponseStatus) {
+    this.httpResponseStatus = httpResponseStatus;
+  }
+
+  public HttpResponseStatus getHttpResponseStatus() {
+    return httpResponseStatus;
+  }
+
+  public int getCode() {
+    return httpResponseStatus.code();
+  }
+
+  public static VeniceReadResponseStatus fromCode(int code) {
+    for (VeniceReadResponseStatus status: values()) {
```

Comment:
We want to avoid calling `values()` because it allocates a new array each time. A few alternatives:

1. You can allocate a static property which contains the result of `values()`, and iterate over the static property here.
2. You can change this function into a switch/case (would need to be manually maintained if new enum values are added, though that is probably not a big deal, and this expectation could be checked trivially via a unit test).

Reply:
> We want to avoid calling values() because it allocates a new array each time.

TIL. 

I like/prefer option 1: static property.